src/main/java/com/thealgorithms/dynamicprogramming/Knapsack.java

```diff
@@ -1,38 +1,40 @@
 package com.thealgorithms.dynamicprogramming;
 
 /**
- * A DynamicProgramming based solution for 0-1 Knapsack problem
+ * A Dynamic Programming based solution for the 0-1 Knapsack problem.
+ * This class provides a method, `knapSack`, that calculates the maximum value that can be
+ * obtained from a given set of items with weights and values, while not exceeding a
+ * given weight capacity.
+ *
+ * @see <a href="https://en.wikipedia.org/?title=0-1_Knapsack_problem">0-1 Knapsack Problem </a>
  */
-public class Knapsack {
+public final class Knapsack {
 
-    private static int knapSack(int W, int[] wt, int[] val, int n) throws IllegalArgumentException {
-        if (wt == null || val == null) {
-            throw new IllegalArgumentException();
+    /**
+     * Solves the 0-1 Knapsack problem using Dynamic Programming.
+     *
+     * @param weightCapacity The maximum weight capacity of the knapsack.
+     * @param weights        An array of item weights.
+     * @param values         An array of item values.
+     * @return The maximum value that can be obtained without exceeding the weight capacity.
+     * @throws IllegalArgumentException If the input arrays are null or have different lengths.
+     */
+    public static int knapSack(final int weightCapacity, final int[] weights, final int[] values) throws IllegalArgumentException {
+        if (weights == null || values == null || weights.length != values.length) {
```

Comment:
Is it really needed to check if `weights` or `values` are `null`?

Reply:
@Manan-09 why did you mark this as resolved?

Reply:
I think it's a reasonable check. There's nothing else to prevent them from being null

Reply:
In this case I would highly recommend adding two more tests where the input arrays are `null` (i.e. one with `weights` being `null` and second with `values` being `null`).